lib/swm050/gpio.c

```diff
@@ -187,11 +185,123 @@ is enabled, GPIO and AF of the SWD pins will be both unavailable.
 */
 void gpio_sel_swd(bool en)
 {
-	if (en) {
-		SWD_SEL = 1;
-	} else {
-		SWD_SEL = 0;
-	}
+	SWD_SEL = en;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets the pins as external interrupts, rather than normal GPIO
+
+Enable interrupts on the selected pins.  If you want to quickly
+switch on and off interrupts, use gpio_int_mask() after calling this.
+
+@param[in] gpios Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_int_en(uint16_t gpios)
+{
+	GPIO_INTEN_A |= gpios;
+}
+
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets bits in the interrupt mask
+
+When interrupts are masked, it prevents them from being received, which is a
+quicker way to turn on and off GPIO interrupts (after calling gpio_int_en()).
+
+@param[in] gpios Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+	     
+@param[in] gpios Pin mask selection @ref gpio_int_masked
+	     Whether to mask or unmask pins.
+*/
+void gpio_int_mask(uint16_t gpios, bool masked)
+{
+	GPIO_INTMASK_A = masked ? (GPIO_INTMASK_A | gpios) : (GPIO_INTMASK_A & ~gpios);
+}
+
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets whether the pins are edge triggered or level triggered
+
+Sets whether the pins are edge triggered or level triggered.  Edge-triggered
+interrupt bits must be cleared by software.
+
+@param[in] gpios Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+
+@param[in] type Trigger Type @ref gpio_trig_type
+	     Level or edge triggered
+*/
+void gpio_int_type(uint16_t gpios, bool type)
+{
+	GPIO_INTLEVEL_A = type ? (GPIO_INTLEVEL_A | gpios) : (GPIO_INTLEVEL_A & ~gpios);
+}
+
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets the interrupt trigger polarity
+
+Sets whether the interrupt is triggered by a high or low level/edge.
+
+@param[in] gpios Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+
+@param[in] pol Polarity @ref gpio_pol 
+	     High or low level/edge
+*/
+void gpio_int_pol(uint16_t gpios, bool pol)
+{
+	GPIO_INTPOLARITY_A = pol ? (GPIO_INTPOLARITY_A | gpios) : (GPIO_INTPOLARITY_A & ~gpios);
```

Comment:
did "make stylecheck" not complain about this?  this isn't saving any code, it just makes the lines harder to read